src/EthereumHelpers.js

```diff
@@ -296,19 +317,25 @@ async function callWithRetry(
  * @param {AbiItem[]} contractABI The ABI of the contract to instantiate.
  * @param {string} [address] The address of the deployed contract; if left
  *        unspecified, the contract won't be pointed to any address.
+ * @param {number} [deployedAtBlock]
  *
  * @return {Contract} A contract for the specified ABI at the specified address,
  *         with default `from` and revert handling set.
  */
-function buildContract(web3, contractABI, address) {
-  /** @type {Contract} */
-  const contract = /** @type {any} */ (new web3.eth.Contract(contractABI))
+function buildContract(web3, contractABI, address, deployedAtBlock) {
+  /** @type {Web3Contract} */
+  const web3Contract = /** @type {any} */ (new web3.eth.Contract(contractABI))
   if (address) {
-    contract.options.address = address
+    web3Contract.options.address = address
   }
-  contract.options.from = web3.eth.defaultAccount || undefined
+  web3Contract.options.from = web3.eth.defaultAccount || undefined
   // @ts-ignore A newer version of Web3 is needed to include handleRevert.
-  contract.options.handleRevert = true
+  web3Contract.options.handleRevert = true
+
+  /** @type {Contract} */
+  const contract = (web3Contract)
+
+  contract.deployedAtBlock = deployedAtBlock || undefined
```

Comment:
I would almost prefer if we set this as an `options` value, tbh. That feels like a more appropriate place for this.